sdks/python/apache_beam/metrics/execution.py

```diff
@@ -106,11 +107,13 @@ def __init__(self, key, committed, attempted):
     self.key = key
     self.committed = committed
     self.attempted = attempted
+    self.committed_only = committed_only
 
   def __eq__(self, other):
     return (
         self.key == other.key and self.committed == other.committed and
-        self.attempted == other.attempted)
+        (self.committed_only or other.committed_only or
+         self.attempted == other.attempted))
```

Comment:
This is not what I meant when I suggested a matcher, see https://github.com/hamcrest/PyHamcrest#writing-custom-matchers for an example.

Reply:
By searching for MatcherBase, found an existing implementation under testing folder. 

Updated.   PTAL.